prime-dotnet-webapi/Controllers/OrganizationsController.cs

```diff
@@ -110,6 +123,115 @@ public async Task<ActionResult> CreateOrganization(CreateOrganizationViewModel c
+
+        // GET: api/Organizations/claim
+        /// <summary>
+        /// Check if organization claim exists by a given search criteria.
+        /// </summary>
+        [HttpGet("claims", Name = nameof(OrganizationClaimExists))]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(ApiResultResponse<OrganizationClaim>), StatusCodes.Status200OK)]
+        public async Task<ActionResult> OrganizationClaimExists([FromQuery] OrganizationClaimSearchOptions search)
+        {
+            // check if the organization exists with the given PEC
+            if (!string.IsNullOrEmpty(search.Pec))
+            {
+                var organization = await _organizationService.GetOrganizationByPecAsync(search.Pec);
+                if (organization == null)
+                {
+                    return BadRequest("Organization does not exist with the passed in site PEC.");
+                }
+            }
+            var result = await _organizationClaimService.OrganizationClaimExistsAsync(search);
+
+            return Ok(result);
+        }
+
+        // GET: api/Organizations/5/claims
+        /// <summary>
+        /// Find OrganizationClaim by Organization ID.
+        /// </summary>
+        [HttpGet("{organizationId}/claims", Name = nameof(GetOrganizationClaimByOrgId))]
+        [Authorize(Roles = Roles.ViewSite)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiResultResponse<OrganizationClaim>), StatusCodes.Status200OK)]
+        public async Task<ActionResult> GetOrganizationClaimByOrgId(int organizationId)
+        {
+            var claim = await _organizationClaimService.GetOrganizationClaimAsync(organizationId);
+            if (claim == null)
+            {
+                return NotFound("No claim by a SigningAuthority exists for given Organization.");
+            }
+            return Ok(claim);
+        }
+
+        // POST: api/Organizations/5/claims/1/approve
+        /// <summary>
+        /// Approve claim for an existing Organization.
+        /// </summary>
+        [HttpPost("{organizationId}/claims/{claimId}/approve", Name = nameof(ApproveOrganizationClaim))]
+        [Authorize(Roles = Roles.EditSite)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        public async Task<ActionResult> ApproveOrganizationClaim(int organizationId, int claimId)
+        {
+            var orgClaim = await _organizationClaimService.GetOrganizationClaimAsync(organizationId);
```

Comment:
I don't see why you would have to get the claim by organizationId when you have the claimId already. I could see passing both IDs into the service method to make sure that the specified claim is actually on the specified organization.